migrator/migrations/m_192_to_m_193_report_config_migrations/migration_impl.go

```diff
@@ -0,0 +1,310 @@
+
+			//if deterministic id exists no need to copy the config
+			//since getMigratedReportConfigIfExsists only reads data from older migration, no need to write a new tx
+			migrated, data, err := getMigratedReportConfigIfExsists(reportConfigProto.GetId(), db, database.DBCtx)
+			if err != nil {
+				return errors.Wrapf(err, "failed to query report config with id %s", reportConfigProto.GetId())
+			}
+			if migrated {
+				if reflect.DeepEqual(reportConfig, data) {
+					log.Infof("Old v1 report config proto %+v is different from v2 copy", data)
+				}
+				continue
+			}
+
+			//since checkifNotifierExsists only reads data from older migration, no need to write a new tx
+			notifierFound, err := checkifNotifierExsists(reportConfigProto.GetEmailConfig().GetNotifierId(), db, database.DBCtx)
+			if err != nil {
+				return errors.Wrapf(err, "failed to query notifier with id %s", reportConfigProto.GetEmailConfig().GetNotifierId())
+			}
+
+			if !notifierFound {
+				log.Infof("Did not find notifier with id %s attached to report config %+v", reportConfigProto.GetEmailConfig().GetNotifierId(),
+					reportConfigProto)
+				log.Infof("Did not find notifier with id %s.converted configs %+v", reportConfigProto.GetEmailConfig().GetNotifierId(),
+					convertedReportConfigs)
+				continue
+			}
+
+			// create v2 report config from v1
+			newConfig := createV2reportConfig(reportConfigProto)
+
+			// create notifier config for report_configuration_notifier table
+			notifierConfig := createNotifier(reportConfigProto)
+			newConfig.Notifiers = append(newConfig.Notifiers, notifierConfig)
+
+			// add notifier to notifier_configurations_notifiers
+			reportNotifierGorm, err := updatedSchema.ConvertNotifierConfigurationFromProto(notifierConfig, 0, newConfig.GetId())
+			if err != nil {
+				log.Errorf("failed to convert notifier config from proto %+v", notifierConfig, err)
+				continue
+			}
+
+			// assign version to 2 to new copy and version to 1 in original so that they are not re-created during migration
+			newConfig.Version = 2
+			reportConfigProto.Version = 1
+			// convert report config proto back to gorm model
+			convertedGormNewConfig, err := updatedSchema.ConvertReportConfigurationFromProto(newConfig)
+			if err != nil {
+				log.Errorf("failed to convert report config from proto %+v", newConfig, err)
+				continue
+			}
+			convertedGormReportConfig, err := updatedSchema.ConvertReportConfigurationFromProto(reportConfigProto)
+			if err != nil {
+				log.Errorf("failed to convert report config from proto %+v", reportConfigProto, err)
+				continue
+			}
+
+			// create report snapshot for last run report job
+			reportSnapshot := createReportSnapshot(reportConfigProto, newConfig)
+			if reportSnapshot != nil {
+				// convert report snapshot to GORM
+				reportSnapshotGORM, err := updatedSchema.ConvertReportSnapshotFromProto(reportSnapshot)
+				if err != nil {
+					log.Errorf("failed to convert snapshot from proto %+v", reportSnapshot, err)
```

Comment:
nit : Error message can be "failed to convert report snapshot from proto to gorm : %+v"